packages/suite-desktop/src-electron/http-receiver.ts

```diff
@@ -0,0 +1,145 @@
+            },
+            /**
+             * Register more routes here. Each route must have pathname and handler function.
+             */
+        ];
+        this.server = http.createServer(this.onRequest);
+    }
+
+    getServerAddress() {
+        const address = this.server.address();
+        // net.AddressInfo may also be string for a server listening on a pipe or Unix domain socket, the name is returned as a string.
+        if (!address || typeof address === 'string') return null;
+
+        return address;
+    }
+
+    getRouteAddress(pathname: any) {
+        const address = this.getServerAddress();
+        const route = this.routes.find(r => r.pathname === pathname);
+        if (!route) return;
+        if (address) {
+            return `http://${address.address}:${address.port}${route.pathname}`;
+        }
+    }
+
+    start() {
+        return new Promise((resolve, reject) => {
+            this.server.on('error', e => {
+                // @ts-ignore - type is missing
+                if (e.code === 'EADDRINUSE') {
+                    // todo:
+                }
+                this.server.close();
+                return reject();
+            });
+            this.server.listen(HttpReceiver.PORTS[0], '127.0.0.1', undefined, () => {
+                const address = this.getServerAddress();
+                if (address) {
+                    this.emit('server/listening', address);
+                }
+                return resolve(address);
+            });
+        });
+    }
+
+    stop() {
+        // note that this method only stops listening but keeps existing connections open and thus port blocked
+        this.server.close();
+    }
+
+    /**
+     * Entry point for handling requests
+     */
+    private onRequest = (request: http.IncomingMessage, response: http.ServerResponse) => {
+        // mostly ts stuff. request should always have url defined.
+        if (!request.url) {
+            this.emit('server/error', 'Unexpected incoming message');
+            return;
+        }
+
+        const { pathname } = url.parse(request.url, true);
+
+        // only method GET is supported
+        if (request.method !== 'GET') {
+            return;
+        }
+
+        const route = this.routes.find(r => r.pathname === pathname);
+        if (route) {
+            // original type has url as optional, Request alias makes it required.
+            return route.handler(request as Request, response);
+        }
+    };
+
+    /**
+     * Handlers sections starts here
+     */
+
+    private oauthHandler = (request: Request, response: http.ServerResponse) => {
+        const { query } = url.parse(request.url, true);
+
+        // send data back to main window
+        if (typeof query.code === 'string') {
+            this.emit('oauth/code', query.code);
+        }
+
+        // todo: would be nice to find how to close default browser window after redirecting.
+        const template = `
```

Comment:
We could serve an html file here with some styles in the future.

Reply:
sure, there should be a button that will focus app on click. but later.